pkg/services/folder/folderimpl/conversions.go

```diff
@@ -0,0 +1,104 @@
+package folderimpl
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	authlib "github.com/grafana/authlib/types"
+	"github.com/grafana/grafana/pkg/apimachinery/utils"
+	"github.com/grafana/grafana/pkg/infra/slugify"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/folder"
+	"github.com/grafana/grafana/pkg/services/user"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func (ss *FolderUnifiedStoreImpl) UnstructuredToLegacyFolder(ctx context.Context, item *unstructured.Unstructured) (*folder.Folder, error) {
+	meta, err := utils.MetaAccessor(item)
+	if err != nil {
+		return nil, err
+	}
+
+	info, _ := authlib.ParseNamespace(meta.GetNamespace())
+	if info.OrgID < 0 {
+		info.OrgID = 1 // This resolves all test cases that assume org 1
+	}
+
+	title, _, _ := unstructured.NestedString(item.Object, "spec", "title")
+	description, _, _ := unstructured.NestedString(item.Object, "spec", "description")
+
+	uid := meta.GetName()
+	url := ""
+	if uid != folder.RootFolder.UID {
+		slug := slugify.Slugify(title)
+		url = dashboards.GetFolderURL(uid, slug)
+	}
+
+	created := meta.GetCreationTimestamp().Time.UTC()
+	updated, _ := meta.GetUpdatedTimestamp()
+	if updated == nil {
+		updated = &created
+	} else {
+		tmp := updated.UTC()
+		updated = &tmp
+	}
+
+	creator, err := ss.getUserFromMeta(ctx, meta.GetCreatedBy())
+	if err != nil {
+		return nil, err
+	}
+
+	updater, err := ss.getUserFromMeta(ctx, meta.GetUpdatedBy())
+	if err != nil {
+		return nil, err
+	}
+	if updater.UID == "" {
+		updater = creator
+	}
```

Comment:
To match legacy behavior: https://github.com/grafana/grafana/blob/suntala/fix-user-login-2/pkg/services/folder/folderimpl/folder.go#L331-L332